hw/ip_templates/racl_ctrl/data/racl_ctrl.hjson.tpl

```diff
@@ -110,6 +113,55 @@
   ],
 
   registers: [
+    % for policy in policies:
+    { name: "POLICY_${policy['name'].upper()}${"_SHADOWED" if enable_shadow_reg else ""}"
+      desc: '''
+            Read and write policy for ${policy['name']}
+            '''
+      swaccess: "rw"
+      hwaccess: "hro"
+    % if enable_shadow_reg:
+      shadowed: "true"
+      update_err_alert: "recov_ctrl_update_err"
+      storage_err_alert: "fatal_fault"
+    % endif
+      fields: [
+        { bits: "31:16"
+          name: "write_perm"
+          resval: ${policy['wr_default']}
+          desc: '''
+                Write permission for policy ${policy['name']}
+                '''
+        }
+        { bits: "15:0"
+          name: "read_perm"
+          resval: ${policy['rd_default']}
+          desc: '''
+                Read permission for policy ${policy['name']}
+                '''
+        }
+      ]
+    }
+    { reserved: "1" }
+    % endfor
+    { skipto: "${hex(2 * 4 * 2 ** nr_role_bits)}" }
```

Comment:
I'm trying to make sure that I understand what this is doing. I think that if there are N possible roles, then this will jump to an address that gives space for 8N registers.

I'm a bit surprised: the number of registers that we generate doesn't seem to have anything that iterates over roles (but we *do* iterate over policies). Is that correct?

Reply:
I changed it further. The registers are added at a fixed offset `0xF4`, independent of the number of policies. THat way we have room more more policies before and match the internal implementation for most of the systems.